linux/deploy_vm/deploy_vm_from_iso.yml

```diff
@@ -213,12 +203,80 @@
         vm_wait_log_delay: 30
         vm_wait_log_retries: 120
 
-    - name: "Wait 60s for OS rebooting"
-      ansible.builtin.pause:
-        seconds: 60
+    - name: "Take a screenshot when autoinstall complete message is received"
+      include_tasks: ../../common/vm_take_screenshot.yml
+      vars:
+        vm_take_screenshot_local_path: "{{ current_test_log_folder }}"
+
+    # VMware Photon OS kickstart file can't add shutdown action,
+    # so here needs to shutdown it separately
+    - name: "Shutdown VMware Photon OS VM after auto install completes"
+      when:
+        - unattend_install_conf is defined
+        - unattend_install_conf is match('Photon')
+      block:
+        - name: "Wait for guest full name on VMware Photon OS VM"
+          include_tasks: ../../common/vm_wait_guest_fullname.yml
+
+        - name: "Get guest IP address"
+          include_tasks: ../../common/update_inventory.yml
+          vars:
+            update_inventory_timeout: 300
+
+        - name: "Shutdown guest OS to remove serial port"
+          include_tasks: ../utils/shutdown.yml
+
+    - name: "Wait for VM being powered off"
+      include_tasks: ../../common/vm_wait_power_state.yml
+      vars:
+        expected_power_status: 'poweredOff'
+        wait_power_state_timeout: 600
+
+    - name: "Delete unattend install ISO"
+      include_tasks: delete_unattend_install_iso.yml
+      when:
+        - transferred_unattend_iso is defined
+        - transferred_unattend_iso
+
+    - name: "Change the boot disk to be first boot device in boot order"
+      include_tasks: ../../common/vm_set_boot_options.yml
+      vars:
+        boot_order_list: ['disk']
+      when: guest_id is match('ubuntu.*')
+
+    - name: "Download serial output file before removing serial port"
```

Comment:
If request #477 is merged, here can use collect_serial_port_log.yml adeed by #477

Reply:
If there is no more comments about this change. I will firstly merge this one, and then update here in https://github.com/vmware/ansible-vsphere-gos-validation/pull/477 together with other unresolved comments.